src/internal_modules/roc_packet/ireader.h

```diff
@@ -23,9 +24,11 @@ class IReader {
     virtual ~IReader();
 
     //! Read packet.
+    //!
     //! @returns
-    //!  next available packet or NULL if there are no packets.
-    virtual PacketPtr read() = 0;
+    //!  0 on success and @p packet is set to next available packet,
+    //!  negative error code otherwise. @see error::ErrorCode.
+    virtual ROC_ATTR_NODISCARD int read(PacketPtr& packet) = 0;
```

Comment:
Let's add a few clarifications:

* never returns positive number
* if code is negative, packet is never set
* if packet is set, code is always zero